lib/s3middleware/azureHelpers/mpuUtils.ts

```diff
@@ -172,19 +171,22 @@ const putNextSubPart = (
     subPartIndex: number,
     resultsCollector: ResultsCollector,
     log: RequestLogger,
-    cb: (err: ArsenalError | null | undefined) => void,
 ) => {
     const { uploadId, partNumber, bucketName, objectKey } = partParams;
     const subPartSize = getSubPartSize(
         subPartInfo, subPartIndex);
     const subPartId = getBlockId(uploadId, partNumber,
         subPartIndex);
     resultsCollector.pushOp();
-    errorWrapperFn('uploadPart', 'createBlockFromStream',
-        client => client.getBlockBlobClient(objectKey)
-            .stageBlock(subPartId, () => subPartStream, subPartSize, {}),
-        (err: azure.RestError | null | undefined) => resultsCollector.pushResult(err, subPartIndex),
-        log, cb);
+    errorWrapperFn('uploadPart', 'createBlockFromStream', async client => {
+        try {
+            const result = await client.getBlockBlobClient(objectKey)
+                .stageBlock(subPartId, () => subPartStream, subPartSize, {});
+            resultsCollector.pushResult(null, subPartIndex);
+        } catch (err: any) {
+            resultsCollector.pushResult(err, subPartIndex);
+        }
+    }, log, () => {});
```

Comment:
I'm not sure i fully understand why we no longer wait for the blocks to be PUT.

it seems to me that we still do asynchronous operations  as we still await for the client to be retrieved ?

I know there's an async alternative to `stageBlock` (`stageBlockAsync`), so i'm assuming this part is synchronous. but not sure about `getBlockBlobClient`.

Reply:
getBlockBlobClient() is synchronous, it just creates the block-blob client object (from the container client) : i.e. fundamentally it just adds the blob name in the url...